sanic/app.py

```diff
@@ -311,19 +316,37 @@ def register_middleware(
             **response** - Invoke before the response is returned back
         :return: decorated method
         """
-        if attach_to == "request":
+        retval = middleware
+        location = MiddlewareLocation[attach_to.upper()]
+
+        if not isinstance(middleware, Middleware):
+            middleware = Middleware(
+                middleware,
+                location=location,
+                priority=priority if isinstance(priority, int) else 0,
+            )
+        elif middleware.priority != priority and isinstance(priority, int):
```

Comment:
Might be worth reversing these statements to make it clear we're comparing two `int`s.
```suggestion
        elif isinstance(priority, int) and middleware.priority != priority:
```

Reply:
Not necessary.